src/test/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImplTest.java

```diff
@@ -0,0 +1,63 @@
+package eu.europeana.metis.sandbox.service.workflow;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import eu.europeana.metis.sandbox.common.exception.ServiceException;
+import eu.europeana.metis.utils.ZipFileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.zeroturnaround.zip.ZipException;
+
+
+@ExtendWith(MockitoExtension.class)
+public class HarvestServiceImplTest {
+
+
+  private final ZipFileReader zipfilereader = new ZipFileReader();
+
+  @InjectMocks
+  private HarvestServiceImpl harvestService;
+
+  @Test
+  void harvestService_expectSuccess() throws IOException {
+
+    Path datasetPath = Paths.get("src", "test", "resources", "zip", "dataset-valid.zip");
+    assertTrue(Files.exists(datasetPath));
+
+    var expectedRecords = zipfilereader.getContentFromZipFile(Files.newInputStream(datasetPath));
```

Comment:
How is that you can make the assumption that `zipfilereader.getContentFromZipFile` will get you the `expectedRecords` ? Is this what the `HttpHarvesterImpl` does underneath ? Because I do not see the `zipfilereader` anywhere else in the code so I guess it is only kept for testing

Reply:
Correct. It is only used for test.